hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/InformationExtractorJdbcDatabaseMetaDataImpl.java

```diff
@@ -6,1017 +6,174 @@
-											"Encountered primary keys differing name on table %s",
-											tableInformation.getName().toString()
-									)
+			extractionContext.getQueryResults(
+					"select * from " + tableName + " where 1=0",
+					null,
+					resultSet -> {
+						final ResultSetMetaData metaData = resultSet.getMetaData();
+						final int columnCount = metaData.getColumnCount();
+
+						for ( int i = 1; i <= columnCount; i++ ) {
+							final String columnName = metaData.getColumnName( i );
+							final ColumnInformationImpl columnInformation = new ColumnInformationImpl(
+									tableInformation,
+									DatabaseIdentifier.toIdentifier( columnName ),
+									metaData.getColumnType( i ),
+									new StringTokenizer( metaData.getColumnTypeName( i ), "() " ).nextToken(),
+									metaData.getPrecision( i ),
+									metaData.getScale( i ),
+									interpretNullable( metaData.isNullable( i ) )
 							);
+							tableInformation.addColumn( columnInformation );
 						}
+						return null;
 					}
-
-					final int columnPosition = resultSet.getInt( "KEY_SEQ" );
-
-					final Identifier columnIdentifier = DatabaseIdentifier.toIdentifier(
-							resultSet.getString( "COLUMN_NAME" )
-					);
-					final ColumnInformation column = tableInformation.getColumn( columnIdentifier );
-					pkColumns.add( columnPosition-1, column );
-				}
-			}
-			finally {
-				resultSet.close();
-			}
-
-			if ( firstPass ) {
-				// we did not find any results (no pk)
-				return null;
-			}
-			else {
-				// validate column list is properly contiguous
-				for ( int i = 0; i < pkColumns.size(); i++ ) {
-					if ( pkColumns.get( i ) == null ) {
-						throw new SchemaExtractionException( "Primary Key information was missing for KEY_SEQ = " + ( i+1) );
-					}
-				}
-
-				// build the return
-				return new PrimaryKeyInformationImpl( pkIdentifier, pkColumns );
-			}
-		}
-		catch (SQLException e) {
-			throw convertSQLException( e, "Error while reading primary key meta data for " + tableInformation.getName().toString() );
-		}
-	}
-
-	@Override
-	public Iterable<IndexInformation> getIndexes(TableInformation tableInformation) {
-		final Map<Identifier, IndexInformationImpl.Builder> builders = new HashMap<>();
-		final QualifiedTableName tableName = tableInformation.getName();
-		final Identifier catalog = tableName.getCatalogName();
-		final Identifier schema = tableName.getSchemaName();
-
-		final String catalogFilter;
-		final String schemaFilter;
-
-		if ( catalog == null ) {
-			catalogFilter = "";
-		}
-		else {
-			catalogFilter = catalog.getText();
-		}
-
-		if ( schema == null ) {
-			schemaFilter = "";
-		}
-		else {
-			schemaFilter = schema.getText();
-		}
-
-		try {
-			ResultSet resultSet = extractionContext.getJdbcDatabaseMetaData().getIndexInfo(
-					catalogFilter,
-					schemaFilter,
-					tableName.getTableName().getText(),
-					false,        // DO NOT limit to just unique
-					true        // DO require up-to-date results
```

Comment:
I just noticed that `true` for the `approximate` argument to `DatabaseMetaData#getIndexInfo says:
```
approximate - when true, result is allowed to reflect approximate or out of data values; when false, results are requested to be accurate
```
ORM was passing `true` here, meaning that approximate data is OK, but the comment is inconsistent saying that up-to-date results are required.

Which is correct? 

FWIW, when `DatabaseMetaData` is not used to get this information, there does not seem to be a way to request up-to-date information. 

For example, MySQL documentation says that `information_schema_stats_expiry=0` should be set in order to get up-to-date statistics. This is probably not what is wanted on a production database.

Reply:
@dreab8, any idea what the intention is here? Is this just a documentation bug?

Reply:
This should not delay merging this issue. I just wanted to raise it since I saw the inconsistency.

Reply:
not sure but I would say we want to use up-to-date info but better to ask @sebersole for a confirmation

Reply:
FYI, I don't see a way for Hibernate to force this for MySQL without setting a system variable.

See https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-statistics-table.html.